### PR TITLE
update code toward php8 compliance

### PR DESCRIPTION
**Pull request by grandoc**

### CODE REVIEW

File: htdocs/accountancy/customer/index.php [hunk 1]
```diff
@@ -47,7 +47,7 @@
 if ($user->socid > 0) {
 	accessforbidden();
 }
-if (empty($user->rights->accounting->bind->write)) {
+if ($user->hasRight('accounting', 'bind', 'write')) {
```
Comment (eldy) [inline htdocs/accountancy/customer/index.php:50]: Test differs

Comment (grandoc) [reply htdocs/accountancy/customer/index.php]: I have push the fix

File: htdocs/accountancy/bookkeeping/listbyaccount.php [hunk 2]
```diff
@@ -401,7 +401,7 @@
 	$uploaddir = $conf->societe->dir_output;
 	include DOL_DOCUMENT_ROOT.'/core/actions_massactions.inc.php';
 
-	if (!$error && $action == 'deletebookkeepingwriting' && $confirm == "yes" && $user->rights->accounting->mouvements->supprimer) {
```
Comment (eldy) [inline htdocs/accountancy/bookkeeping/listbyaccount.php:404]: Test differs

File: htdocs/accountancy/bookkeeping/listbyaccount.php [hunk 3]
```diff
@@ -401,7 +401,7 @@
...
+	if (!$error && $action == 'deletebookkeepingwriting' && $confirm == "yes" && !$user->hasRight('accounting', 'mouvements', 'supprimer')) {
```
Comment (eldy) [inline htdocs/accountancy/bookkeeping/listbyaccount.php:404]: Test differs

Comment (grandoc) [reply htdocs/accountancy/bookkeeping/listbyaccount.php]: I made a fix

**Pull request merged**